pydm/widgets/baseplot.py

```diff
@@ -65,6 +70,10 @@ def __init__(self, color=None, lineStyle=None, lineWidth=None, **kws):
         if color is not None:
             self.color = color
 
+        if hasattr(self, "channels"):
+            self.destroyed.connect(functools.partial(widget_destroyed,
+                                                     self.channels))
```

Comment:
There seems to be a bit of confusion where things are an attribute/property vs a method. `.channels` seems like it should be the former to me...